lib/matplotlib/tests/test_collections.py

```diff
@@ -830,6 +831,56 @@ def test_collection_set_verts_array():
         assert np.array_equal(ap._codes, atp._codes)
 
 
+def _fig_to_bytes(fig):
+    with io.BytesIO() as bio:
+        fig.savefig(bio)
+        return bio.getvalue()
+
+
+@pytest.fixture()
+def ind():
+    return np.linspace(16, 365, (365-16)*4)
+
+
+@pytest.fixture()
+def dep1(ind):
+    return np.sin(2*np.pi*ind/153) + np.cos(2*np.pi*ind/127)
+
+
+@pytest.fixture()
+def dep2(dep1):
+    return dep1 + .2
+
+
+@pytest.mark.parametrize("fname", ["fill_between", "fill_betweenx"])
+def test_fbp_collection_set_data(fname, ind, dep1, dep2):
+    dep3 = dep2.copy()
+    dep3[100], dep3[200] = dep3[2], dep3[200]
+
+    fig1, ax1 = plt.subplots()
+    getattr(ax1, fname)(ind, dep1, dep2)
+    fig1b = _fig_to_bytes(fig1)
+
+    fig2, ax2 = plt.subplots()
+    coll2 = getattr(ax2, fname)(ind, dep1, dep3)
+
+    assert fig1b != _fig_to_bytes(fig2)
```

Comment:
This test is not needed: You test that `fill_between` with different data yields different figures. But the existing tests for fill_between already validate that you get a figure with correct visualization of the data. That different data yield different figures is a corrollary of this.

Reply:
db3af6447c354a4ced461af8482939f627343557